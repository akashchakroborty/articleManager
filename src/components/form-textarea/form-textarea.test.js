import React from "react";
import { shallow } from "enzyme";
import FormTextarea from "./form-textarea.component";

describe("FormTextarea component", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      label: "Description",
      value: "mock value",
      handleChange: mockHandleChange,
    };

    wrapper = shallow(<FormTextarea {...mockProps} />);
  });

  it("should render FormTextarea component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange method when input changes", () => {
    wrapper.find("FormTextareaContainer").simulate("change");

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("should render FormTextareaLabel if there is a label", () => {
    expect(wrapper.exists("FormTextareaLabel")).toBe(true);
  });

  it("should not render FormTextareaLabel if there is no label", () => {
    const mockNewProps = {
      label: "",
      value: "mock value",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormTextarea {...mockNewProps} />);

    expect(newWrapper.exists("FormTextareaLabel")).toBe(false);
  });

  it("should render FormTextareaLabel with no shrink class when value is empty", () => {
    const mockNewProps = {
      label: "title",
      value: "",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormTextarea {...mockNewProps} />);

    expect(newWrapper.find("FormTextareaLabel").hasClass("")).toBe(true);
  });

  it("should render FormTextareaLabel with shrink class when there is any value", () => {
    const mockNewProps = {
      label: "title",
      value: "mock",
      handleChange: mockHandleChange,
    };

    const newWrapper = shallow(<FormTextarea {...mockNewProps} />);

    expect(newWrapper.find("FormTextareaLabel").hasClass("shrink")).toBe(true);
  });
});
