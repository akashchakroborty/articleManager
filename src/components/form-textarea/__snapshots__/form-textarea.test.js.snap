// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextarea component should render FormTextarea component 1`] = `ShallowWrapper {}`;
